### DISCUSSION TITLE
Cannot get field name/value from PDF text field (Angular 15.2.1)

### DISCUSSION THREAD

**Discussion by haroldopaulino** [stephanrauh/ngx-extended-pdf-viewer#1736] [Q&A]
I have a PDF document with 2 fields, where first is for a person's name and the next is a date.

Upon submitting the form, the date field is received from the formData, however the person's name field is not there in the first few formFilled events.

If I don't refresh the page and reopen the document so many times and fill the fields and submit over and over again, eventually the person's name does make it to the formData.

I have updated the package with:
npm update ngx-extended-pdf-viewer

This issue does not happen to all PDF documents, just to some of them.

Somehow, when some of our clients compose their PDF documents and upload them to our system, this issue happens.

I've downloaded an affected document to my computer and I can edit the fields and save the document without issues using Mac's  Preview.

Any ideas?

Thank you for your time,
Harold.

![Screenshot 2023-04-03 at 8 20 05 AM](https://user-images.githubusercontent.com/28379115/229538327-f3fa62b4-1002-471b-a07d-d44bfa01044c.png)

**Comment by stephanrauh** [comment]
Strange. Do you see any message in the console window? I wonder if there's an exception. Judging from your description, it also might be a timing issue or a race condition.

**Reply by haroldopaulino** [reply]
Thank you for your timely response. I really appreciate that!

Since the document has sensitive information, I've tried editing with Adobe Acrobat and removing the sensitive words, but when I saved it, the issue doesn't happen anymore.

It's like the PDF document gets "fixed" when I save it.

I'm trying multiple possibilities with Adobe Acrobat and Microsoft Word, with exporting and playing with options for the export.

As far as the console goes, here are some screenshots.

<img width="1478" alt="Screenshot 2023-04-03 at 3 38 10 PM" src="https://user-images.githubusercontent.com/28379115/229633682-96210c65-8855-4408-969e-f8f77dbe8bd8.png">

<img width="1331" alt="Screenshot 2023-04-03 at 3 03 22 PM" src="https://user-images.githubusercontent.com/28379115/229633750-a9b21e28-700b-4fcb-b72b-6dbe6e8df1b8.png">

**Reply by stephanrauh** [reply]
That rings a bell. Could your issue be related to #1708?

**Reply by haroldopaulino** [reply]
I think it is related, because on that ticket the field value was also null.

Since you were able to reproduce the issue, in what scenario does the PDF document "cause" the problem to happen? I ask because only certain PDF documents are affected.

Do you have any idea when that fix will be ready for production?

**Reply by stephanrauh** [reply]
No idea. I'm afraid most evenings I only manage to answer tickets and questions. So the open tickets start piling up disturbingly. Can I talk you into supporting me? Maintaining a library is much more fun if it's not just me but a team!

**Reply by haroldopaulino** [reply]
I can give it a shot, sure.

**Comment by fadel2ab** [comment]
@haroldopaulino I try with several versions (16, 17 and 18) but I still have the same problem (https://github.com/stephanrauh/ngx-extended-pdf-viewer/discussions/1708). I would like to know if you managed to find a solution to this problem ?

**Reply by haroldopaulino** [reply]
@fadel2ab, I'm sorry you're having issues.

I ended up using pdf.js straight up.

This library extension seems to be maintained by a single individual, as there is too much for that person to keep up with.

I saw no reason to stick around, and I have to admit that pdf.js is working very well.

**Reply by stephanrauh** [reply]
That's why I keep asking people to contribute to the project. To be honest, I'm puzzled nobody wants to do so. My other open source project, BootsFaces, recruited contrubitors every couple of months (and lost them almost as quickly). I wonder if that's one of the differences between the JavaScript community and the Java folks?
